swayfmt/src/utils/map/newline.rs

```diff
@@ -270,22 +283,31 @@ fn insert_after_span(
     newline_sequence: NewlineSequence,
     formatted_code: &mut FormattedCode,
     threshold: usize,
-) -> Result<usize, FormatterError> {
-    let mut sequence_string = String::new();
-    write!(
-        sequence_string,
-        "{}",
-        format_newline_sequence(&newline_sequence, threshold)
-    )?;
+) -> Result<i64, FormatterError> {
+    let sequence_string = format_newline_sequence(&newline_sequence, threshold);
+    let mut len = sequence_string.len() as i64;
     let mut src_rope = Rope::from_str(formatted_code);
 
+    // Remove the previous sequence_length, that will be replaced in the next statement
+    let mut remove_until = at;
+    for i in at..at + newline_sequence.sequence_length {
+        if src_rope.get_char(i) != Some('\n') {
+            break;
+        }
+        remove_until = i;
+    }
+    if remove_until > at {
+        let _ = src_rope.try_remove(at..remove_until);
```

Comment:
We shouldn't be ignoring `Result`s, generally, unless there's a good reason for it. In this case if you are trying to avoid a `panic` then maybe the length shouldn't change if you're unable to remove it?
Can you try changing this to:
```rust
if let Err(_) = src_rope.try_remove(at..remove_until) {
    len -= (remove_until - at) as i64;
}
```
I may not have the full context here, feel free to ignore this if my suggestion doesn't work.